crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -320,23 +320,29 @@ pub const RocStr = extern struct {
         return (ptr - 1)[0] == utils.REFCOUNT_ONE;
     }
 
-    pub fn asSlice(self: RocStr) []u8 {
+    pub fn asSlice(self: *const RocStr) []const u8 {
         return self.asU8ptr()[0..self.len()];
     }
 
-    pub fn asSliceWithCapacity(self: RocStr) []u8 {
+    pub fn asSliceWithCapacity(self: *const RocStr) []const u8 {
         return self.asU8ptr()[0..self.getCapacity()];
     }
 
-    pub fn asU8ptr(self: RocStr) [*]u8 {
+    pub fn asSliceWithCapacityMut(self: *RocStr) []u8 {
+        return self.asU8ptrMut()[0..self.getCapacity()];
+    }
 
-        // Since this conditional would be prone to branch misprediction,
-        // make sure it will compile to a cmov.
-        // return if (self.isSmallStr()) (&@bitCast([@sizeOf(RocStr)]u8, self)) else (@ptrCast([*]u8, self.str_bytes));
+    pub fn asU8ptr(self: *const RocStr) [*]const u8 {
```

Comment:
In the body, we were returning a pointer to what is semantically a stack value.

I believe that in practice, zig will always pass a `RocStr` by-reference, which is why the old signature with `self: RocStr` worked. But now either zig or LLVM is stricter, and that is not valid any more (and I'd think that it never really was). So we take the string by-reference explicitly